### ISSUE TITLE
Custom Exception handler and HttpStatusException support

### ISSUE THREAD

**Issue by aravindavk** [trikko/serverino#21]
## Custom Exception Handler Support:

```d
// If no connection or any other Db error below function can raise DbException
void listNotes()
{
    auto query = "SELECT * FROM notes";
    auto rs = conn.execute(query);
    return rs.serializeTo!Note;
}

@endpoint @route!"/handler1"
void handler1(Request req, Output res)
{
    try
    {
        auto notes = getNotes;
        ...
    }
    catch (DbException)
    {
        res.status = 500;
        res.write("Failed to get notes");
    }
}

@endpoint @route!"/handler2"
void handler2(Request req, Output res)
{
    try
    {
        auto notes = getNotes;
        ...
    }
    catch (DbException)
    {
        res.status = 500;
        res.write("Failed to get notes");
    }
}
```

Adding try catch in every handler that uses the Db functions is repetitive. Is it possible to provide a custom exception handler in settings so that all custom exceptions can be handled from one place.

- Vibe.d reference: https://vibed.org/api/vibe.http.server/HTTPServerSettings.errorPageHandler
- Handy Httpd reference: https://andrewlalis.github.io/handy-httpd/guide/handling-exceptions.html#fatal-errors

## HttpStatusException support

To halt from the handler, throwing exception is very easy way compared to checking the return statements of each function calls.

Example without HttpStatusException support:

```d
bool enforceHttp(Output res, bool cond, string message, ushort status = 400)
{
    if (!cond)
    {
        res.status = status;
        res.write(message);
        return false;
    }

    return true;
}

void handler(Request req, Output res)
{
    immutable string name = req.post.read("name", "");
    immutable string email = req.post.read("email", "");
    
    if (!ensureHttp(res, !name.empty, "Name should not be empty"))
        return;

    if (!ensureHttp(res, email.isEmail, "Invalid Email format"))
        return;
    ...
}
```

Example with HttpStatusException support:

```d
void ensureHttp(bool cond, string message, ushort status = 400)
{
    if (!cond)
        throw HttpStatusException(status, message);
}

void handler(Request req, Output res)
{
    immutable string name = req.post.read("name", "");
    immutable string email = req.post.read("email", "");
    ensureHttp(!name.empty, "Name should not be empty");
    ensureHttp(email.isEmail, "Invalid Email format");

    ...
}
```

**Comment by trikko** [comment]
Probably I'm going to add

```d
@onWorkerException bool handler(Request r, Output o, Exception e)
{
    // If handled, return true
}
```

**Comment by trikko** [comment]
Fixed with commit c669ff2
